nodeJS/APIs/APIs.md

```diff
@@ -12,41 +12,42 @@ By the end of this lesson, you should be able to do the following:
 
-Typically this takes the form of 2 URI's per resource, one for the whole collection and one for a single object in that collection, for example, you might get a list of blog-posts from `/posts` and then get a specific post from `/posts/:postid`. You can also nest collections in this way. To get the list of comments on a single post you would access `/posts/:postid/comments` and then to get a single comment: `/posts/:postid/comments/:commentid`.
+|Verb   | Action | Example |
+---     | ---    | ---
+|POST   | Create | `POST /posts` Creates a new blog post |
+|GET    | Read   | `GET /posts/:postid` Fetches a single post |
+|PUT    | Update | `PUT /posts/:postid` Updates a single post |
+|DELETE | Delete | `DELETE /posts` Deletes a single post |
 
-As mentioned above, you access and manipulate things using the HTTP verbs that you already know about, GET, POST, PUT and DELETE. The usage here is basically what you would expect, but to be clear, let's work through a few examples.
 
-- To get a list of blog posts you would `GET` `/posts`, so in your express app you would respond to  `app.get("/posts")` with a list of all posts.
-- To add a new blog-post you would `POST` to `/posts`.
-- To update the contents of a blog-post you would `PUT` to the URI of the specific post you're updating. `/posts/:postid`
-- To delete a post you would `DELETE` `/posts/:postid`
+Each part of an API URI specifies the resource. For example, `GET /posts` would return the entire list of blog posts while `GET /posts/:postid` specifies the exact blog post we want. We could nest further with `GET /posts/:postid/comments` to return a list of comments for that blog post or even `GET /posts/:id/comments/:commentid` for a very specific blog post comment.
 
 ### CORS
 The [Same Origin Policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) is an important security measure that basically says "Only requests from the same origin (the same IP address or URL) should be allowed to access this API". (Look at the link above for a couple of examples of what counts as the 'same origin'.) This is a big problem for us because we are specifically trying to set up our API so that we can access it from different origins, so to enable that we need to set up Cross-origin resource sharing, or CORS.
 
-Setting up CORS in express is very easy, there's a middleware that does the work for us. [This article](https://medium.com/@alexishevia/using-cors-in-express-cac7e29b005b) explains everything you need to do, and the official docs can be found [here](https://expressjs.com/en/resources/middleware/cors.html).
+Setting up CORS in Express is very easy, there's a middleware that does the work for us. The official docs can be found [here](https://expressjs.com/en/resources/middleware/cors.html#enabling-cors-pre-flight).
```

Comment:
Shouldn't be necessary to link to a Medium article for using Express middleware, the official docs do a fine job IMO. Linked to the specific subsection for clarity and it'd be more instructional to read actual documentation.